### PR TITLE
Migrate Sort (MatrixNms-8, MulticlassNms-8, NonMaxSuppression-5)

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Create template plugin reference tests for target operation: MatrixNms-8, MulticlassNms-8, NonMaxSuppression-5*
 - *Remove ngraph backend files*
 - *Update CMakeLists.txt*
 - *Add the following regex to skip_tests_config.cpp:*
 - *R"(.\*ReferenceMatrixNmsTest.\*esiType=i64.\*evoType=i64.\*)"*
 - *R"(.\*ReferenceMulticlassNmsTest.\*esiType=i64.\*evoType=i64.\*)"*
 - *R"(.\*ReferenceNonMaxSuppressionTest.\*esiType=i32.\*evoType=i32.\*)"*

### Tickets:
 - *64119*
 - *64121*
 - *64096*

### CODE REVIEW

**Comment by steve-y** [comment]
@iefode @ilyachur skip_tests_config.cpp contains the following regex because they generated the following exception and error:

- CVS-64119 MatrixNms : `R"(.\*ReferenceMatrixNmsTest.\*esiType=i64.\*evoType=i64.*)"`
When MatrixNms's output type is set to i64, it generates the following exception:
```
C++ exception with description "
inference-engine/src/inference_engine/src/ie_common.cpp:81 map::at" thrown in the test body.
```
And checking at line 333 in `inference-engine/src/inference_engine/src/cpp_interfaces/interface/ie_iplugin_internal.cpp`
shows the testing values:
Actual values:
```
outputsInfo[0] - "MatrixNms_1280.0"
outputsInfo[1] - "MatrixNms_1280/convert.1"
outputsInfo[2] - "MatrixNms_1280/convert.2"
```
Expected values:
```
outputsInfo[0] - "MatrixNms_1280.0"
outputsInfo[1] - "MatrixNms_1280.1"
outputsInfo[2] - "MatrixNms_1280.2"
```
- CVS-64121 MulticlassNms : `R"(.\*ReferenceMulticlassNmsTest.\*esiType=i64.\*evoType=i64.\*)"`
When MulticlassNms's output type is set to i64, it generates the following exception:
```
C++ exception with description "
inference-engine/src/inference_engine/src/ie_common.cpp:81 map::at" thrown in the test body.
```
And checking at line 333 in `inference-engine/src/inference_engine/src/cpp_interfaces/interface/ie_iplugin_internal.cpp`
shows the testing values:
Actual values:
```
outputsInfo[0] - "MulticlassNms_1280.0.0"
outputsInfo[1] - "MulticlassNms_1280.0/convert.1"
outputsInfo[2] - "MulticlassNms_1280.0/convert.2"
```
Expected values:
```
outputsInfo[0] - "MulticlassNms_1280.0.0"
outputsInfo[1] - "MulticlassNms_1280.0.1"
outputsInfo[2] - "MulticlassNms_1280.0.2"
```
- CVS-64096 NonMaxSuppression : `R"(.\*ReferenceNonMaxSuppressionTest.\*esiType=i32.\*evoType=i32.\*)"`
When NonMaxSuppression's output type is set to i32, it generates the following failures:
```
../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
Expected equality of these values:
  refBlob.get_element_type()
    Which is: i32
  outBlob.get_element_type()
    Which is: i64
../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
Expected equality of these values:
  refBlob.get_element_type()
    Which is: i32
  outBlob.get_element_type()
    Which is: i64
```
- They seem to have output type i32 or i64 but not both.

**Comment by ilyachur** [comment]
> @iefode @ilyachur skip_tests_config.cpp contains the following regex because they generated the following exception and error:
> 
>     * CVS-64119 MatrixNms : `R"(.\*ReferenceMatrixNmsTest.\*esiType=i64.\*evoType=i64.*)"`
>       When MatrixNms's output type is set to i64, it generates the following exception:
> 
> 
> ```
> C++ exception with description "
> inference-engine/src/inference_engine/src/ie_common.cpp:81 map::at" thrown in the test body.
> ```
> 
> And checking at line 333 in `inference-engine/src/inference_engine/src/cpp_interfaces/interface/ie_iplugin_internal.cpp` shows the testing values: Actual values:
> 
> ```
> outputsInfo[0] - "MatrixNms_1280.0"
> outputsInfo[1] - "MatrixNms_1280/convert.1"
> outputsInfo[2] - "MatrixNms_1280/convert.2"
> ```
> 
> Expected values:
> 
> ```
> outputsInfo[0] - "MatrixNms_1280.0"
> outputsInfo[1] - "MatrixNms_1280.1"
> outputsInfo[2] - "MatrixNms_1280.2"
> ```
> 
>     * CVS-64121 MulticlassNms : `R"(.\*ReferenceMulticlassNmsTest.\*esiType=i64.\*evoType=i64.\*)"`
>       When MulticlassNms's output type is set to i64, it generates the following exception:
> 
> 
> ```
> C++ exception with description "
> inference-engine/src/inference_engine/src/ie_common.cpp:81 map::at" thrown in the test body.
> ```
> 
> And checking at line 333 in `inference-engine/src/inference_engine/src/cpp_interfaces/interface/ie_iplugin_internal.cpp` shows the testing values: Actual values:
> 
> ```
> outputsInfo[0] - "MulticlassNms_1280.0.0"
> outputsInfo[1] - "MulticlassNms_1280.0/convert.1"
> outputsInfo[2] - "MulticlassNms_1280.0/convert.2"
> ```
> 
> Expected values:
> 
> ```
> outputsInfo[0] - "MulticlassNms_1280.0.0"
> outputsInfo[1] - "MulticlassNms_1280.0.1"
> outputsInfo[2] - "MulticlassNms_1280.0.2"
> ```
> 
>     * CVS-64096 NonMaxSuppression : `R"(.\*ReferenceNonMaxSuppressionTest.\*esiType=i32.\*evoType=i32.\*)"`
>       When NonMaxSuppression's output type is set to i32, it generates the following failures:
> 
> 
> ```
> ../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
> Expected equality of these values:
>   refBlob.get_element_type()
>     Which is: i32
>   outBlob.get_element_type()
>     Which is: i64
> ../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
> Expected equality of these values:
>   refBlob.get_element_type()
>     Which is: i32
>   outBlob.get_element_type()
>     Which is: i64
> ```
> 
>     * They seem to have output type i32 or i64 but not both.

Could we understand the root cause of the issue? Is it a plugin limitation or the reference implementation issue? Because template plugin is a small wrapper to call reference implementations.

**Comment by steve-y** [comment]
In MatrixNms and MulticlassNms cases, they seem to be with plugin issue because they seem to be related to `Core::compile_model()` in that function, when the model is converted to network, the name is changed with unwanted suffix.

In NonMaxSuppression case, it seems to be with reference implementation because the failure is occurred in the Validation and the outBlob is generated in i64 only.

**Review by iefode — APPROVED** [review]

**Comment by ilyachur** [comment]
> In MatrixNms and MulticlassNms cases, they seem to be with plugin issue because they seem to be related to `Core::compile_model()` in that function, when the model is converted to network, the name is changed with unwanted suffix.
> 
> In NonMaxSuppression case, it seems to be with reference implementation because the failure is occurred in the Validation and the outBlob is generated in i64 only.

Could you create tickets for issues?

File: docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+        const ov::element::Type output_type,
+        const bool sort_result_across_batch,
+        const bool normalized,
+        const Tensor& boxes, const Tensor& scores,
+        const Tensor& expectedSelectedScores, const Tensor& expectedSelectedIndices,
+        const Tensor& expectedValidOutputs, const std::string& testcaseName = "") :
+        nms_top_k(nms_top_k),
+        score_threshold(score_threshold),
+        sort_result_type(sort_result_type),
+        keep_top_k(keep_top_k),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp:34]: Can we use MatrixNMS::Attributes instead of this set of parameters?

**Comment by steve-y** [comment]
> > In MatrixNms and MulticlassNms cases, they seem to be with plugin issue because they seem to be related to `Core::compile_model()` in that function, when the model is converted to network, the name is changed with unwanted suffix.
> > In NonMaxSuppression case, it seems to be with reference implementation because the failure is occurred in the Validation and the outBlob is generated in i64 only.
> 
> Could you create tickets for issues?

Okay, I will.

File: docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp (hunk 1 above)
Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp:34]: I tried it but  MatrixNMS::Attributes has no constructor so it's almost same as this set of parameters.

Comment (ilyachur) [reply docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp:34]: The function will have less number of arguments. So I propose to use the structure

Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp]: As MatrixNMS::Attributes needs a constructor, use local Attributes structure.

File: docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+
+#include "openvino/op/matrix_nms.hpp"
+#include "openvino/op/constant.hpp"
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct Attributes {
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp:15]: Why do you need this internal structure? Can we reuse operation Attributes?

File: docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp (hunk 1 above)
Comment (ilyachur) [reply docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp]: Can we use initializer list to init operation Attribtes?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**